### PR TITLE
RELATED: RAIL-3627 Add support for individual insight exports

### PR DESCRIPTION
**Pull request by no23reason**
* add support for individual insight widget exports
* unify the logic in export drill insight as well

The OptionsMenu cannot be customized which is needed by gdc-dashboards (due to Explore from here functions). This will be added in a follow up PR.

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4921/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6579/ : SUCCESS in 18m 34s

File: libs/sdk-ui-dashboard/src/presentation/dashboardContexts/ExportDialogContext.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        setDialogConfig(config);
+    }, []);
+
+    const closeDialog = useCallback(() => {
+        setIsOpen(false);
+    }, []);
+
+    return (
+        <ExportDialogContext.Provider
+            value={{
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/dashboardContexts/ExportDialogContext.tsx:52]: Use memo for ExportDialogContext.Provider value

File: libs/sdk-ui-dashboard/src/presentation/dialogs/ExportDialogProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import React, { useCallback } from "react";
+import { useExportDialogContext } from "../dashboardContexts";
+
+/**
+ * @internal
+ */
+export const ExportDialogProvider: React.FC = () => {
+    const { closeDialog, dialogConfig, isOpen } = useExportDialogContext();
+
+    const onClose = useCallback(() => {
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/dialogs/ExportDialogProvider.tsx:12]: Pls extract variable for dep. first,

const {onClose: dialogConfigOnClose} = dialogConfig;

than use it in dep. array

File: libs/sdk-ui-dashboard/src/presentation/widget/common/errorUtils.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    isDataTooLargeToDisplay,
+    isProtectedReport,
+    isEmptyAfm,
+    isNegativeValues,
+];
+
+/**
+ * @internal
+ */
+export const isDataError = (error: GoodDataSdkError | undefined): boolean => {
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/presentation/widget/common/errorUtils.ts:48]: Is posible create union type of DataError ??? or use combineGuards from utils to get ReturnType ? 
And than set return valu as proper guard? 

export const isDataError = (error: GoodDataSdkError | undefined): error is DataErrorType =>

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4927/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6583/ : SUCCESS in 14m 38s

File: libs/sdk-ui-dashboard/src/presentation/widget/common/errorUtils.ts (hunk 3 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/presentation/widget/common/errorUtils.ts]: updated

File: libs/sdk-ui-dashboard/src/presentation/dialogs/ExportDialogProvider.tsx (hunk 2 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/presentation/dialogs/ExportDialogProvider.tsx:13]: updated

File: libs/sdk-ui-dashboard/src/presentation/dashboardContexts/ExportDialogContext.tsx (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/presentation/dashboardContexts/ExportDialogContext.tsx]: Good catch, udpated

**Review by ivanmjartan — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1668

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1400/ : FAILURE in 2m 45s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1677/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/461/ : SUCCESS in 18m 49s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6586/ : SUCCESS in 14m 25s

**Pull request merged**